### PR TITLE
feat(persistence): add the ability to serialize the model when needed

### PR DESCRIPTION
**Pull request by Lutra-Fs**

### CODE REVIEW

**Review by LorenzoValentine — APPROVED** [review]

File: src/services/model/modelsave.json
Comment (peipacut) [inline src/services/model/modelsave.json:1]: this should probably be removed

**Review by peipacut — APPROVED** [review]

File: tests/services/model/modelService.test.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+test("serialise empty input tensor", async() => {
+    expect(ser(new MockModelService(new Float32Array())))
+        .toBe(JSON.stringify(`
+
+        `))
+})
+
+test("serialise valid model", async() => {
+    let model = new MockModelService(new Float32Array([1,2,3,4,5.5,6.6]));
```
[the remark this answers is not in the document]
Comment (Lutra-Fs) [reply tests/services/model/modelService.test.ts]: @peipacut

File: tests/services/model/modelService.test.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+async function mockModelFactory(
+  _a: string,
+  _b: [number, number] = [64, 64],
+  _c: number,
+  _d: number,
+  _e: number,
+  _f: number,
+): Promise<ModelService> {
+    return Promise.resolve(new MockModelService()) // eslint-disable-line
```
[the remark this answers is not in the document]
Comment (Lutra-Fs) [reply tests/services/model/modelService.test.ts:127]: @peipacut

**Review by peipacut — APPROVED** [review]

File: tests/services/model/modelService.test.ts [hunk 3]
```diff
@@ -117,13 +117,16 @@ test("serialise valid model", async() => {
...
+  _gridisize: [number, number] = [64, 64],
+  _batchsize: number,
+  _channelsize: number,
+  _outputchannelsize: number,
+  _fpslim: number,
 ): Promise<ModelService> {
+    // unlike conventional model services, this one doesn't require
+    // a worker to construct so no async await is used
+    // we'll ignore the required awaited promise error in this case
     return Promise.resolve(new MockModelService()) // eslint-disable-line
```
Comment (Lutra-Fs) [inline tests/services/model/modelService.test.ts:130]: @peipacut  How about like this
```suggestion
    return await new MockModelService(); // eslint-disable-line
```

Comment (Lutra-Fs) [reply tests/services/model/modelService.test.ts:130]: nvm

**Pull request merged**